### ISSUE TITLE
Rule suggestion - add configuration to PascalCase naming of classes and interfaces

### ISSUE THREAD

**Issue by carpben** [palantir/tslint#4487]
**Is your rule for a general problem or is it specific to your development style?**
I think it could be relevant to others. 

**What does your suggested rule do?**
Add configuration options to the PascalCase naming to classes and interfaces. 

**List several examples where your rule could be used**

I have this final variable. 
```ts
export const SEARCH_MODES: I_SEARCH_MODES = {
   ONSITE: "onsite" ,
   ONLINE: "online",
}
```
I thought it would be helpful if the name and the case of the type could convey that all it's fields are final. So I named it  
```ts
export interface I_SEARCH_MODES {
   readonly ONSITE: "onsite"
   readonly ONLINE: "online"
}
```
but this yields a warning since it doesn't follow camelcase.  I wanted to allow upper case for interfaces, but there is no such option. 
Perhaps we can add a configuration that allows upper case for interfaces.

**Comment by TokugawaTakesi** [comment]
Underscore in class names is also will be useful in cases like this:

+ `class PersonModel__RawApi {} `
+ `class PersonModel__DefaultImplementation {}`
+ `class PersonModel__Customized {}`

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
